### ISSUE TITLE
Implement link previews for web+* URIs using fedilinks fallback protocol handlers

### ISSUE THREAD

**Issue by SoniEx2** [matrix-org/synapse#16542]
Fallback protocol handlers, as defined here: https://fedilinks.org/2 should be used to provide link previews for web+* URIs.

This would help promote web+* URIs, which are a limited form of composable URIs designed for compatibility with existing computer systems. Because we believe links should be able to work on a best-effort basis, with fallbacks to other protocols as needed. This would help with protocol proliferation. "Protocols, not platforms."

**Comment by clokep** [comment]
Providing an example link is usually helpful with this sort of thing.

**Comment by SoniEx2** [comment]
real world implementation

link: <web+ap://portend.place/users/smitten> (filtered by github)

fallback handler: https://portend.place/.well-known/protocol-handler?target=web%2Bap://portend.place/users/smitten (this is what would be used for the preview)

**Comment by clokep** [comment]
FTR the fallback handler is just a 302 redirect to `location: /api/v1/akkoma/protocol-handler?target=web%2Bap%3A%2F%2Fportend.place%2Fusers%2Fsmitten`

**Comment by SoniEx2** [comment]
yes, usually there are 2 redirects involved. here's another another one:

link: <web+ap://is-a.cat/users/ar>

fallback handler: https://is-a.cat/.well-known/protocol-handler?target=web%2Bap://is-a.cat/users/ar

**Comment by SoniEx2** [comment]
since this would be a 2-line patch, do you think you (matrix) can first get rid of your real name policy, just as Linux has done?

**Comment by clokep** [comment]
> since this would be a 2-line patch

I suspect we'd like to see a bit more uptake in the ecosystem before supporting this.

> do you think you (matrix) can first get rid of your real name policy, just as Linux has done?

You can sign-off privately if you would prefer: https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#private-sign-off

**Comment by reivilibre** [comment]
> since this would be a 2-line patch, do you think you (matrix) can first get rid of your real name policy, just as Linux has done?

we also accept common law names (names claimed by legitimate usage or repute), see https://matrix-org.github.io/synapse/latest/development/contributing_guide.html?highlight=common%20law#sign-off.
